sky/provision/docker_utils.py

```diff
@@ -192,32 +173,103 @@ def _run(self,
+                            break
+                        else
+                            exec su -l $USER
+                            sleep 5
+                        fi
+                    done
+                    if [[ $docker_ready -eq 0 ]]
+                    then
+                        echo "docker_ready:" "N"
+                    fi
+
+                    # check runtime info
+                    echo "docker_runtime:" $({self.docker_cmd} info -f "{{{{.Runtimes}}}}")
+
+                    if [[ $_self_initialized -eq 0 ]]
+                    then 
+                        # check container status
+                        container_status=$({self.docker_cmd} ps -a --filter name="{self.container_name}" --format "{{{{.Status}}}}")
+                        echo "container_status:" $container_status
+                        # if container is up, check its image
+                        if [[ $container_status == *"Up"* ]]
+                        then
+                            container_image=$({self.docker_cmd} inspect --format='{{{{.Config.Image}}}}' {self.container_name})
+                            echo "container_image:" $container_image
+                        fi
+                    fi 
+                fi
+                echo "status_checking_completed:" "Y"
+            """)
+        ret = {}
+
+        for l in result.split('\n'):
+            if ':' in l:
+                x = l.split(':')
+                k = x[0].strip()
+                v = ':'.join(x[1:]).strip()
+                if v in ['Y', 'N']:
+                    v = True if v == 'Y' else False  # type: ignore
+                if k == 'MemAvailable':
+                    k = 'mem_available_in_kb'
+                    v = int(v.split()[0])  # type: ignore
+                ret[k] = v
+        return ret
+
     def initialize(self) -> str:
         specific_image = self.docker_config['image']
 
-        self._check_docker_installed()
+        status = self._check_host_status()
+        logger.info(f'Host status: {str(status)}')
 
-        # SkyPilot: Check if the container is exited but not removed.
-        # If true, then we can start the container directly.
-        # Notice that we will skip all setup commands, so we need to
-        # manually start the ssh service.
-        if self._check_container_exited():
-            self.initialized = True
-            self._run(f'{self.docker_cmd} start {self.container_name}')
-            self._run('sudo service ssh start', run_env='docker')
-            return self._run('whoami', run_env='docker')
+        if 'container_status' in status:
+            if 'Exited' in status['container_status']:
+                logger.info('Container is exited but not removed.')
+                self.initialized = True
+                self._run(f'{self.docker_cmd} start {self.container_name}')
+                self._run('sudo service ssh start', run_env='docker')
+                return self._run('whoami', run_env='docker')
```

Comment:
Can this be integrated into the `check_host_status` command as well?

Reply:
It's merged to the action ssh below.